### PR TITLE
NFC: Move RegAlloc headers to include dir

### PR DESCRIPTION
**Pull request by optimisan**

### CODE REVIEW

**Comment by optimisan** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/119672?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#119540** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/119540?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#119672** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/119672?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/119672?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#119181** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/119181?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#118462** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/118462?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#117309** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/117309?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: llvm/include/llvm/CodeGen/InterferenceCache.h [hunk 1]
```diff
@@ -142,7 +142,7 @@ class LLVM_LIBRARY_VISIBILITY InterferenceCache {
 
   // Point to an entry for each physreg. The entry pointed to may not be up to
   // date, and it may have been reused for a different physreg.
-  unsigned char* PhysRegEntries = nullptr;
+  unsigned char *PhysRegEntries = nullptr;
```
Comment (arsenm) [inline llvm/include/llvm/CodeGen/InterferenceCache.h:145]: Reformat shouldn't be done at same time as the move

**Review by arsenm — COMMENTED** [review]
Why? These are supposed to be private headers to the regalloc implementation

File: llvm/include/llvm/CodeGen/InterferenceCache.h (hunk 1 above)
Comment (optimisan) [reply llvm/include/llvm/CodeGen/InterferenceCache.h:145]: Yes I can put the new greedy pass in a header instead.

**Pull request closed without merging**